server/db.js

```diff
@@ -9,51 +10,63 @@ let db = new pouchDb("peachworks", { adapter: "memory" });
 const getRecipes = (req, res) => {
   //TODO: move this out into a cron job
   fetchRecipesFromPeachworks(() => {
-    db.get("recipes").then(doc => {
-      res.json(doc.recipes);
-    });
+    db
+      .allDocs({
+        startkey: "recipe-",
+        endkey: "recipe-\uffff",
+        include_docs: true
+      })
+      .then(result => {
+        const allRecipes = result.rows.map(row => row.doc);
+        res.json({ recipes: allRecipes });
+      });
   });
 };
 
 //endpoint which fetches from db for recipe data
 const getRecipe = (req, res) => {
-  const id = req.params.id;
+  const id = parseInt(req.params.id, 10);
 
   //TODO: move this out into a cron job
   fetchRecipeFromPeachworks(id, () => {
     db.get("recipe-" + id).then(doc => {
-      res.json(doc.recipe);
+      res.json(doc);
     });
   });
 };
 
-const createOrUpdate = (doc, callback) => {
+const createOrUpdate = (newDoc, callback) => {
   db
-    .get(doc._id)
-    .then(_doc => {
-      doc._rev = _doc._rev;
-      db.put(doc);
+    .upsert(newDoc._id, storedDoc => {
```

Comment:
ah! they did implement upsert! I saw a mention of it when I looked up create or update, but didn't see it in the pouchdb api docs